etl_manager/extract_metadata.py

```diff
@@ -0,0 +1,409 @@
+import cx_Oracle
+
+from etl_manager.utils import write_json, read_json
+
+
+def create_database_connection(settings_file):
+    """Connects to an Oracle database, with settings taken from the specified json file
+
+    The json file should specify:
+        - host: usually localhost
+        - password
+        - port: usually 1521
+        - service_name
+        - user
+
+    Returns a cx_Oracle database connection.
+    """
+    db_settings = read_json(settings_file)
+
+    dsn = cx_Oracle.makedsn(
+        host=db_settings["host"],
+        port=db_settings["port"],
+        service_name=db_settings["service_name"],
+    )
+    connection = cx_Oracle.connect(
+        user=db_settings["user"],
+        password=db_settings["password"],
+        dsn=dsn,
+        encoding="UTF-8",
+    )
+    return connection
+
+
+def get_table_names(database: str, connection) -> list:
+    """Gets names of tables in the database
+
+    You can save the output into a file, or pass it directly to create_table_json
+
+    Parameters
+    ----------
+    database (str):
+        The name of the database/schema - for example DELIUS_ANALYTICS_PLATFORM
+
+    Returns
+    -------
+    List of table names as strings
+
+    """
+    cursor = connection.cursor()
+    cursor.execute(f"SELECT table_name FROM all_tables WHERE OWNER = '{database}'")
+    result = cursor.fetchall()
+    names = [r[0] for r in result]
+    return names
+
+
+def create_json_for_database(
+    description: str, name: str, bucket: str, base_folder: str, location="./metadata",
```

Comment:
Think this needs to be `meta_data` to be consistent with `etl_manager` naming. Unless we changed it? (I know we discussed it but not sure we implemented it)

Reply:
Looks like it is still like that in some other places in etl_manager, so I've changed it. I've also made a note about making a ticket to change it to metadata throughout.